src/storage/src/hummock/local_version_manager.rs

```diff
@@ -230,10 +230,10 @@ impl LocalVersionManager {
             conflict_detector.set_watermark(newly_pinned_version.max_committed_epoch);
         }
 
-        let mut new_version = old_version;
+        let mut new_version = old_version.clone();
```

Comment:
We only need to update the `pinned_version` in local version here so we don't need to clone and replace the whole local version.

```
{
            let mut guard = RwLockUpgradableReadGuard::upgrade(old_version);
            guard.set_pinned_version(newly_pinned_version);
            RwLockWriteGuard::unlock_fair(guard);
}
```

How about this?

Reply:
We do not hope to run `set_pinned_version` with a write lock because it will also change `BTreeMap`.

Reply:
The differences are:
1. We don't need to acquire a write lock if the version id is not larger.
2. We release the write lock before `version_update_notifier_tx.send`.
3. We ensure fair unlock

Given that the version id check is very light-weight, I don't think we can benefit from 1). I am not sure about 2) either. I suspect the necessity of #3620 but maybe 3) is the key point?

Reply:
> We do not hope to run `set_pinned_version` with a write lock because it will also change `BTreeMap`.

I see your point now. LGTM.

Reply:
No....It is not enought. Our goal in https://github.com/singularity-data/risingwave/pull/3620 is to reduce the time holding write mutex